client.go

```diff
@@ -1386,7 +1386,12 @@ func (l *link) muxHandleFrame(fr frameBody) error {
 
 		// Unblock receivers waiting for message disposition
 		if l.receiver != nil {
-			l.receiver.inFlight.remove(fr.First, fr.Last, nil)
```

Comment:
This was always sending nil on the channel, hiding the disposition error to the receiver.
I'm not checking for settlementMode because there shouldn't be any error coming back if the settlemend mode isn't Second, but I might be wrong.